### PR TITLE
Ban org.ow2.asm:asm in non-deployment modules 

### LINKED ISSUE #493: Ban org.ow2.asm:asm in non-deployment modules
Before https://github.com/quarkiverse/quarkus-cxf/pull/487, we were banning `org.ow2.asm:asm` in some modules. We should investigate whether we should/can ban it after https://github.com/quarkiverse/quarkus-cxf/pull/487 and why.

When banned, it would have to be excluded from the follwing BOM entries:

```
[ERROR]     io.quarkiverse.cxf:quarkus-cxf:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-features-logging:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-features-metrics:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-transports-http-hc5:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     org.apache.cxf:cxf-rt-ws-security:3.5.3:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     org.apache.cxf:cxf-rt-ws-mex:3.5.3:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-ws-security:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
```

As a first step, we can do the above exclusions and check whether some tests fail.

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #493

### CODE REVIEW

**Comment by ppalaga** [comment]
@shumonsharif @famod except that you are welcome to review as usual, I'd vote for releasing quarkus-cxf 1.5.0.CR1 right after merging this PR. WDYT?

**Comment by ppalaga** [comment]
I triggered the release of 1.5.0.CR1 in https://github.com/quarkiverse/quarkus-cxf/pull/509 to be able to integrate in Camel Quarkus. 

I'd like to release 1.5.0 on Thursday or Friday after we fix any glitches and upgrade to Quarkus 2.12.0.Final that is scheduled for Aug 24 (tomorrow). WDYT?

File: extensions/pom.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                <artifactId>maven-enforcer-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>enforce-non-deployment</id>
+                        <configuration>
+                            <skip>${enforce-non-deployment.skip}</skip>
+                            <rules>
+                                <bannedDependencies>
+                                    <excludes>
+                                        <!-- <exclude>org.ow2.asm:asm</exclude> -->
```
Comment (famod) [inline extensions/pom.xml:58]: Is it meant to be like that? Doesn't look right to me.

File: extensions/pom.xml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        <module>features-metrics</module>
+        <module>ws-security</module>
+        <module>transports-http-hc5</module>
+    </modules>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.codehaus.gmaven</groupId>
+                <artifactId>groovy-maven-plugin</artifactId>
```
Comment (famod) [inline extensions/pom.xml:28]: YMMV, but a few years back I had bad experiences with this plugin, especially with concurrent builds. Might all be resolved now, don't know.

Anyway, FTR, I think we could do it via a profile that is activated if `src/main/resources/META-INF/quarkus-extension.yaml` exists.

File: extensions/pom.xml (hunk 1 above)
Comment (ppalaga) [reply extensions/pom.xml:58]: Indeed, good catch, thanks!

File: extensions/pom.xml (hunk 2 above)
Comment (ppalaga) [reply extensions/pom.xml:28]: Good idea, let me try it like that.

Comment (ppalaga) [reply extensions/pom.xml:28]: Done in https://github.com/quarkiverse/quarkus-cxf/pull/513

File: extensions/pom.xml (hunk 1 above)
Comment (ppalaga) [reply extensions/pom.xml:58]: DOne in https://github.com/quarkiverse/quarkus-cxf/pull/513

**Pull request merged**